src/vision/tenets.md

```diff
@@ -0,0 +1,20 @@
+# Design tenets for async
+
+The design tenets describe the key principles that drive our work on async. Hopefully, we are able to achieve and honor all of them all of the time. Sometimes, though, they come into conflict, and we have to pick -- in that case, we prefer the tenet earlier in the list.
+
+1. **Minimal overhead.** Rust Async I/O performance should compare favourably with any other language. In the extreme case, it should be possible to use async/await without any required allocation, although this is unlikely to be a common case in production systems.
+2. **Async is like sync, but with blocking points clearly identified.** At the highest level, writing a simple program using asynchronous I/O in Rust should be analogous to writing one that uses synchronous I/O, except that one adds `async` in front of function declarations and adds `.await` after each call. We should aim for analogous design between synchronous and asynchronous equivalents. Similarly, streams should be like asynchronous iterators. One should be able to use the same sort of combinators with streams and to iterate over them in analogous ways.
+3. **Easy to get started, but able to do anything you want.** We should make it simple to write Async I/O code and get things that work reasonably well, but it should be possible for people to obtain fine-grained control as needed.
```

Comment:
Personally I think this should move to spot 2, overtaking "async is like sync...".

I think the "async is like sync" of point two is a bit of an oversimplification, but overall I agree with it as a goal. However the "able to do anything you want" part of this point should take priority to allow more advanced use-cases to keep using shared definitions (triats etc.) from std lib, otherwise the ecosystem will likely split into "the standard" and many specialised runtimes all doing their own thing.